### ISSUE TITLE
Don't use gc.get_referrers for no good reasons in ephemeral on pypy

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5254]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#5254|
|Type|defect|
|Created|2011-09-06 14:39:11Z|
|Branch|https://github.com/twisted/twisted/tree/ephemeral-getstate-pypy-5254|

gc.get_referrers works on pypy, but is super-slow (for good reasons), if there are a lot of objects.

Attachments:

* [don-use-referrers.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6f0/6f05604eb5806dbcd1d006d31cea9a094d8b41d8/e7fa7bdd2e98243617837f506792200a2cef0a20.diff) (870 bytes) - added by fijal on 2011-09-06 14:39:57Z - 
* [dont-use-referrers.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6f0/6f05604eb5806dbcd1d006d31cea9a094d8b41d8/0aca1c158a93f69bc94876b5f8296876244d3a38.diff) (874 bytes) - added by fijal on 2011-09-06 16:41:44Z - the actually working patch

<details><summary>Searchable metadata</summary>

```
trac-id__5254 5254
type__defect defect
reporter__fijal fijal
priority__normal normal
milestone__ 
branch__branches_ephemeral_getstate_pypy_5254 branches/ephemeral-getstate-pypy-5254
branch_author__fijal fijal
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1315319951000000 1315319951000000
changetime__1315420270000000 1315420270000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [32517]) Branching to 'ephemeral-getstate-pypy-5254'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [32518]) Apply dont-use-referrers.diff 

refs [#5254](https://github.com/twisted/twisted/issues/5254)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @exarkun|
|-|-|

I filed [#5257](https://github.com/twisted/twisted/issues/5257) for having a nicer way to detect PyPy.  This looks good for now though, thanks!  If [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/ephemeral-getstate-pypy-5254) look good I'll merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [32519]) Merge ephemeral-getstate-pypy-5254

Author: fijal
Reviewer: exarkun
Fixes: [#5254](https://github.com/twisted/twisted/issues/5254)

On PyPy, skip logging the referrers to a `twisted.persisted.styles.Ephemeral`
instance when it is serialized, because `gc.get_referrers` is prohibitively
expensive on PyPy, because `Ephemeral` is not widely used anymore anyway, and
because the extra information isn't likely to be critically important
anyway.
